sklearn_numba_dpex/kmeans/kernels/kmeans_plusplus.py

```diff
@@ -230,5 +234,9 @@ def kmeansplusplus_single_step(
 
             dpex.barrier(dpex.CLK_LOCAL_MEM_FENCE)
 
-    global_size = (math.ceil(n_samples / work_group_size)) * (work_group_size)
-    return kmeansplusplus_single_step[global_size, work_group_size]
+    global_size = (
+        math.ceil(math.ceil(n_samples / window_n_candidates) / candidates_window_height)
+        * candidates_window_height,
```

Comment:
Can you give this a name?

Reply:
I agree but I don't really now how to it clear, it's twisted that the shape of the grid is adapted to the sliding window on centroids, but each work item apply to one single input sample.